base/compiler/abstractinterpretation.jl

```diff
@@ -885,6 +885,7 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter, resul
         add_backedge!(res.const_result.mi, sv, invoketypes)
         return res
     end
+    f isa InvokeCall && (f = f.f) # unwrap `invoke`-ed function (the call will be virtualized)
```

Comment:
I don't usually like wrapper objects in non-escaped contexts. E.g. one could probably end up with an `InvokeCall` here, by doing something like `const foo = CC.InvokeCall(...)` and then calling foo, but I suppose that wouldn't actually be a problem here, since that would still give the right answer, so I suppose it's not a problem.

Reply:
Okay, I stopped passing the wrapper object through the `f` argument, but rather pass the `invoke` information as an additional argument.

Reply:
Great, I like that better

Reply:
I like that better too